client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -653,6 +656,20 @@ export default {
 
         this.blocks.splice(this.editor.index, 1, block)
         this.unsavedBlocks.add(fetchID(block))
+
+        // Add the block that is a parent block is a tab to unsaved blocks registry0
+        this.blocks.forEach((block) => {
```

Comment:
You dont need to do this, you can just add the parent block when you delete a tab.